services/src/main/java/org/keycloak/services/resources/account/AccountRestService.java

```diff
@@ -469,6 +456,65 @@ public LinkedAccountsResource linkedAccounts() {
         return new LinkedAccountsResource(session, request, client, auth, event, user);
     }
 
+    @Path("/applications")
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @NoCache
+    public Response applications() {
+        checkAccountApiEnabled();
+        auth.requireOneOf(AccountRoles.VIEW_APPLICATIONS, AccountRoles.VIEW_PROFILE);
+
+        List<String> inUseClients = getInUseClients();
+
+        List<ClientRepresentation> apps = new LinkedList<>();
+        for (ClientModel client : realm.getClients()) {
+            if (client.isBearerOnly() || client.getBaseUrl() == null) {
+                continue;
+            }
+            apps.add(modelToRepresentation(client, false, false, inUseClients));
+        }
+
+        for (ClientModel client : new UserSessionManager(session).findClientsWithOfflineToken(realm, user)) {
+            apps.add(modelToRepresentation(client, false, true, inUseClients));
+        }
+
+        for (UserConsentModel consent : session.users().getConsents(realm, user.getId())) {
+            ClientModel client = consent.getClient();
+
+            if (!new StorageId(client.getId()).isLocal()) {
+                apps.add(modelToRepresentation(client, true, false, inUseClients));
+            }
+        }
```

Comment:
I believe the current approach is not correct (but maybe I'm wrong ;)).
1. `realm.getClients()` returns all clients in a realm, so we cannot do this:
https://github.com/keycloak/keycloak/blob/00f7faf2151eb64b5e158c35922a707203875832/services/src/main/java/org/keycloak/services/resources/account/AccountRestService.java#L474
We're basically setting `userConsentRequired` and `offlineAccess` to false for all clients.
1. We're adding some of the clients to the list again here:
https://github.com/keycloak/keycloak/blob/00f7faf2151eb64b5e158c35922a707203875832/services/src/main/java/org/keycloak/services/resources/account/AccountRestService.java#L478
and here:
https://github.com/keycloak/keycloak/blob/00f7faf2151eb64b5e158c35922a707203875832/services/src/main/java/org/keycloak/services/resources/account/AccountRestService.java#L485
If I'm correct, this means some of the clients will be returned twice or even three times.
1. What about the clients that have both `userConsentRequired` and `offlineAccess`? I believe we don't cover this scenario?